GetStream/stream-chat-android
# [3714] Add lifecycle publisher tests

Closes #3714 
### 🎯 Goal

We need to write tests for lifecycle publishers to verify their behavior.

### 🛠 Implementation details

- Added tests for lifecycle publishers.

### 🧪 Testing

- Run the sample app and nothing should be changed in that. 
- Make sure our test cases cover every case. 

### ☑️Contributor Checklist

#### General

#### Code & documentation
